src/domain/youthProfile/helpers/prepareArrayFieldChanges.ts

```diff
@@ -0,0 +1,31 @@
+type Model = { id: string };
+
+type ArrayFieldChanges<Add, Update, Remove> = {
+  add: Add;
+  update: Update;
+  remove: Remove;
+};
+
+function prepareArrayFieldChanges<C, M extends Model>(
```

Comment:
It isn't entirely obvious what this function does. Is it here only to ensure that typing is consistent between form and api or is there something else going on?

Reply:
In the form we just have an array of `additionalContactPersons`. But when we send them to the API, we have to distinct between new ones, ones that are updates and ones that are being removed.

Form data model
```
additionalContactPersons: [
    {...}
]
```
API data model
```
addAdditionalContactPersons: [],
updateAdditionalContactPersons: [],
removeAdditionalContactPersons: [],
```

So this function uses the previous values and the next values to understand which `additionalContactPerson` is new, which is updated and which is removed. But it attempts to be more generic than that, so that this functionality does not need to be recoded on a model per model basis.